app/models/hold.rb

```diff
@@ -27,7 +27,7 @@ def previous_active_holds
   end
 
   def ready_for_pickup?
-    previous_active_holds.empty?
+    previous_active_holds.empty? #&& item.loans.last.ended?
```

Comment:
I agree that extending this method is the way to go here. I _think_ a simpler way to handle this would be:

```suggestion
    previous_active_holds.empty? && item.available?
```

This has the nice side benefit of handling tools that aren't uniquely numbered (these are the ones where `item.borrow_policy.uniquely_numbered == false`).

Reply:
Gotcha. Thank you for the suggestion. I'll try that again. I think I encountered some test weirdness when I tried this first.

Reply:
@jim Maybe you can help me understand why using the `available` method, we get this failure. https://app.circleci.com/pipelines/github/rubyforgood/circulate/660/workflows/ca3ea2bf-f788-46c5-b8a1-9bf15cd124eb/jobs/3483/parallel-runs/0/steps/0-112

I'm puzzled. I step through it, but don't see it. I wonder if there's something with the scope checked_out_exclusive_loan that's used. WDTY?

Reply:
Or maybe it's a data problem and we need to use a slightly different factory like `ended_loan` or `nonexclusive_loan` - I think I tried nonexclusive_loan and was able to get a passing test, but I don't know enough about loan exclusivity to make an informed choice.

Reply:
I re-ran the tests on CI, and they passed, and then everything was looking fine locally until I ran the entire suite in the same order as it was on CI when it failed: `rails test --seed 31967`. Then I was able to reproduce the failure.

Reply:
weird. thank you for investigating. maybe that points to a test data problem then.

Reply:
For anyone following along, here is the followup issue: https://github.com/rubyforgood/circulate/issues/382